torchvision/models/quantization/mobilenet.py

```diff
@@ -0,0 +1,69 @@
+from torch import nn
+from torchvision.models.utils import load_state_dict_from_url
+from torchvision.models.mobilenet import InvertedResidual, ConvBNReLU, MobileNetV2, model_urls
```

Comment:
Do we have plans to add pre-trained models here? Wondering if we should wait until the pre-trained models are available before enabling `pretrained=True` flag.

Reply:
This requires some discussion: For quantization, we want to usually start from a floating point model and then quantize it (for both post training and quantization aware training use cases).  There is also the option of having a pre-trained quantized model.  However, that becomes backend specific. We can do this in stages: For this PR, it will be good to have only the option of loading a pre-trained float model. Should we call it pretrained_float_model=True instead?

Reply:
I have made the change with pretrained_float_model=True. Please let me know of your thoughts

Reply:
I think we should raise an error if we pass `pretrained=True` in this case.

For the reference training scripts, we could do something like

```python
model = torchvision.models.__dict__[model_arch](pretrained=False)
# need to have a mapping between model arch and file it is,
# but it's ok as it's temporary?
model_urls = torchvision.models.__dict__['resnet'].model_urls
state_dict = torchvision.model.utils.load_state_dict_from_url(model_url[model_arch])
model.load_state_dict(state_dict)
```
Thoughts?

Reply:
What we want to cover in this PR is to enable users to start with a pretrained model (this is a float model and is identical to existing models) and allow for users to control it. So, in the reference scripts, we could just do:
```python
model = torchvision.models.quantization.__dict__[model_arch](pretrained_float_model=True)
# model has floating point weights and activations, can be converted to a quantized model using the 
# APIs in torch.quantization
```
I was proposing calling this flag pretrained_float_model because once we have quantized models ready, , we also want to be able to do something like:
```python
model = torchvision.models.quantization.__dict__[model_arch](pretrained_quantized_model=True, backend=args.backend)
# Loads model with pretrained quantized weights, model uses quantized modules from
# torch.nn.quantized. Model is ready for use
```

Reply:
> Do we have plans to add pre-trained models here? Wondering if we should wait until the pre-trained models are available before enabling `pretrained=True` flag.

For this PR, we are adding support for quantizing all float models that are specified in the files that we add: i,e all resnets in resnet.py etc.

Reply:
> I was proposing calling this flag pretrained_float_model because once we have quantized models ready, , we also want to be able to do something like:

I would rather only have the `pretrained` flag, instead of both `pretrained_float_model` and `pretrained_quantized_model`. The reason being that it's easier to add arguments than it is to remove them (due to BC-breaking considerations), and from my understanding there is limited use for the `pretrained_float_model`, which could also be obtained with something like
```python
model = torchvision.models.quantized.__dict__[model_arch](pretrained=False)
model.load_state_dict(
    torchvision.models.__dict__[model_arch](pretrained=True).state_dict())
```
and if more clear than another flag which just copies the state dict from a different model.

> For this PR, we are adding support for quantizing all float models that are specified in the files that we add: i,e all resnets in resnet.py etc.

Sure, but as I mentioned before, except if we are planning on releasing pre-trained models for all those models, we should start with just the ones we are sure we will have the bandwidth to train and release the models. As before, it's easier to add models than it is to remove them.

Reply:
Sure, let me do that. We will support all resnet/mobilenet/shufflenet/googlenet/inception models and can for now say that pretrained=True will cause an error. In the training code, I can do what you suggest.